test-jest/component/Tooltip.spec.tsx

```diff
@@ -0,0 +1,95 @@
+import { render } from '@testing-library/react';
+import React from 'react';
+
+import { Area, AreaChart, Tooltip } from '../../src';
+import { mockMouseEvent } from '../helper/mockMouseEvent';
+
+describe('<Tooltip />', () => {
+  const data = [
+    { name: 'Page A', uv: 400, pv: 2400, amt: 2400 },
+    { name: 'Page B', uv: 300, pv: 1398, amt: 2400 },
+    { name: 'Page C', uv: 200, pv: 9800, amt: 2400 },
+    { name: 'Page D', uv: 278, pv: 3908, amt: 2400 },
+    { name: 'Page E', uv: 189, pv: 4800, amt: 2400 },
+  ];
+
+  test('Without an event, the wrapper is rendered but not visible', () => {
+    const { container } = render(
+      <div role="main" style={{ width: '400px', height: '400px' }}>
+        <AreaChart width={400} height={400} data={data}>
+          <Area dataKey="uv" />
+          <Tooltip />
+        </AreaChart>
+      </div>,
+    );
+
+    const wrapper = container.querySelector('.recharts-tooltip-wrapper');
+    expect(wrapper).toBeInTheDocument();
+    expect(wrapper).not.toBeVisible();
+  });
+
+  test('No content is rendered without an explicit event', () => {
+    const { container } = render(
+      <div role="main" style={{ width: '400px', height: '400px' }}>
+        <AreaChart width={400} height={400} data={data}>
+          <Area dataKey="uv" />
+          <Tooltip />
+        </AreaChart>
+      </div>,
+    );
+
+    const tooltipContentName = container.querySelector('.recharts-tooltip-item-name');
+    const tooltipContentValue = container.querySelector('.recharts-tooltip-item-value');
+    expect(tooltipContentName).toBeNull();
+    expect(tooltipContentValue).toBeNull();
+  });
+
+  test('Mouse over renders content', () => {
+    const { container } = render(
+      <div role="main" style={{ width: '400px', height: '400px' }}>
+        <AreaChart width={400} height={400} data={data}>
+          <Area dataKey="uv" />
+          <Tooltip />
+        </AreaChart>
+      </div>,
+    );
+
+    const chart = container.querySelector('.recharts-wrapper');
+    const mouseOverEvent = mockMouseEvent('mouseover', chart!, { pageX: 200, pageY: 200 });
+    mouseOverEvent.fire();
+
+    // After the mouse over event over the chart, the tooltip wrapper still is not set to visible,
+    // but the content is already created based on the nearest data point.
+    const tooltipContentName = container.querySelector('.recharts-tooltip-item-name');
+    const tooltipContentValue = container.querySelector('.recharts-tooltip-item-value');
+    expect(tooltipContentName).toBeInTheDocument();
+    expect(tooltipContentValue).toBeInTheDocument();
+  });
```

Comment:
If you have any idea / suggestion on how to trigger visibility, I would love to include that in the PR. @ckifer @proke03

Reply:
I think we may need to refactor the Tooltip to use React Portals (theres a FR for that somewhere) in order for this to be easier to test.

I'm okay with merging this and coming back to it

Reply:
For now, I think this is the best way.
I tried implementing that code as a userEvent, but it failed.